elasticapm/transport/base.py

```diff
@@ -54,74 +53,93 @@ def __init__(
         self._max_flush_time = max_flush_time
         self._max_buffer_size = max_buffer_size
         self._queued_data = None
-        self._queue_lock = threading.Lock()
+        self._event_queue = compat.queue.Queue(maxsize=100)
```

Comment:
wondering if this should be configurable.

@axw is it in the Go agent? Also, how did you determine the (default) size of the queue (or ring buffer in your case IIRC)

Reply:
I have separate queues for transactions, spans, and errors; mostly for type-safety. Each queue has a capacity of 1000. I don't think I was very scientific about the choice, just eyeballed a high-throughput test app to ensure it didn't drop a significant amount.

Originally I made them configurable, but the number of queues and buffers increased to the point that I thought it would be unhelpful to expose it initially.

The Go agent has this initial queue, then a ring buffer, and then finally the gzip request buffer. I think that's all of them ;)